k2/addons/k2-serial/zserial/connection.py

```diff
@@ -27,29 +27,48 @@ class SerialConnectionError(Exception):
 
 class SerialConnectionWrapper(object):
 
-    def __init__(self, thread, connection):
+    def __init__(self, thread, connection, serial, protocol_factory):
         self.thread = thread
         self.connection = connection
+        self.serial = serial
+        self.protocol_factory = protocol_factory
 
     def close(self):
         self.thread.close()
 
+    def suspend(self):
+        self.thread.stop()
+        if self.thread.is_alive():
+            raise SerialConnectionError('Failed to suspend serial thread')
+
+    def resume(self):
+        self.thread = ReaderThread(self.serial, self.protocol_factory)
+        self.connection = self.thread.__enter__()
+
+    def is_suspended(self):
+        return False if self.thread.is_alive() else True
```

Comment:
`return not self.thread.is_alive()`
?

Reply:
Can't remember why I did it like that, but your suggestion seem to work now so I'll go with that.